docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -348,10 +348,14 @@ Default contexts are only available for arguments of type `Directory` and `File`
 
 The default context is resolved as follows:
 - If an absolute path is specified:
-   - in a Git repository (defined by the presence of a `.git` sub-directory), the default context is the root of the Git repository.
-   - in a non-repository location (defined by the absence of a `.git` sub-directory), the default context is the directory containing a `dagger.json` file.
+   - in a Git repository (defined by the presence of a `.git` sub-directory), resolution starts from the root of the Git repository.
+   - in a non-repository location (defined by the absence of a `.git` sub-directory), resolution starts from the directory containing a `dagger.json` file.
 - If a relative path is specified
-  - the default context is always the directory containing a `dagger.json` file.
+  - resolution always starts from the directory containing a `dagger.json` file.
+
+:::important
+For security reasons, it is not possible to retrieve files or directories outside the default context's root directory.
```

Comment:
This is conflating a few terms. To clarify:
- **default path**: a path within the _context directory_, to load a directory/file as a default value
- **context directory**: repo root if in a git repo, or the _root directory_ otherwise
- **root directory**: the directory where `dagger.json` is
- **source directory**: the sub-directory within the _root directory_, used by the SDK (`source` field in `dagger.json`)

So there's no "default context". This is more correct:
```suggestion
For security reasons, it is not possible to retrieve files or directories outside the context directory.
```
But the terminology may be a leaking implementation detail if there's a simpler way to explain this (without conflicting with current terminology).

Reply:
I've modified this entire section using the terms "context directory" and "default path". I've not mentioned "source directory" anyway as it did not seem relevant to this section, but let me know in case I got that wrong. I've also updated the example table.